src/hotspot/share/opto/vtransform.cpp

```diff
@@ -902,44 +892,43 @@ VTransformApplyResult VTransformLoadVectorNode::apply(VTransformApplyState& appl
     }
   }
 
-  LoadVectorNode* vn = LoadVectorNode::make(opc, ctrl, mem, adr, adr_type, vlen, bt,
+  LoadVectorNode* vn = LoadVectorNode::make(sopc, ctrl, mem, adr, _adr_type, vlen, bt,
                                             control_dependency());
   DEBUG_ONLY( if (VerifyAlignVector) { vn->set_must_verify_alignment(); } )
   register_new_node_from_vectorization_and_replace_scalar_nodes(apply_state, vn);
   return VTransformApplyResult::make_vector(vn, vlen, vn->memory_size());
 }
 
 VTransformApplyResult VTransformStoreVectorNode::apply(VTransformApplyState& apply_state) const {
+  int     sopc = scalar_opcode();
+  uint    vlen = vector_length();
   StoreNode* first = nodes().at(0)->as_Store();
-  uint  vlen = nodes().length();
-  Node* ctrl = first->in(MemNode::Control);
+  Node* ctrl = apply_state.transformed_node(in_req(MemNode::Control));
+  // first has the correct memory state, determined by VTransformGraph::apply_memops_reordering_with_schedule
   Node* mem  = first->in(MemNode::Memory);
-  Node* adr  = first->in(MemNode::Address);
-  int   opc  = first->Opcode();
-  const TypePtr* adr_type = first->adr_type();
+  Node* adr  = apply_state.transformed_node(in_req(MemNode::Address));
```

Comment:
There is still minimal reliance on nodes / first: but only for mem state. And currently, we cannot remove that yet, because we rely on the memory graph being reordered before vectorization, see VTransformGraph::apply_memops_reordering_with_schedule.

In a future RFE, I will refactor scheduling, so that we build the memory graph during apply.
See step 3 in [plan overfiew.](https://bugs.openjdk.org/browse/JDK-8340093)